src/workers/event.rs

```diff
@@ -870,8 +953,31 @@ mod test {
         assert_eq!(providers.len(), 2, "Should only be 2 providers in state");
         assert_provider(&providers, &provider2, &[&host1_id, &host2_id]);
 
-        // TODO(thomastaylor312): This is just a reminder that if we add in host state updating on
-        // provider/actor events, we should test that here before we hit a host heartbeat
+        // Check that hosts got updated properly
+        let hosts = store.list::<Host>(lattice_id).await.unwrap();
+        assert_eq!(hosts.len(), 2, "Should only have 2 hosts");
```

Comment:
General note that I essentially moved a few assertions _before_ the host heartbeat to ensure that the new functionality worked as well as leaving some assertions after the host heartbeat to ensure we handle the heartbeat properly too.